script/import_scripts/yahoogroup.rb

```diff
@@ -0,0 +1,150 @@
+
+# Import YahooGroups data as exported into MongoDB by:
+#   https://github.com/jonbartlett/yahoo-groups-export
+
+class ImportScripts::YahooGroup < ImportScripts::Base
+
+  def initialize
+    super
+
+    client = Mongo::Client.new([ '192.168.10.1:27017'], :database => 'syncro')
+    db = client.database
+    Mongo::Logger.logger.level = Logger::FATAL
+    puts "connected to db...."
+
+    @collection = client[:posts]
+
+    @user_profile_map = { }
+
+  end
+
+  def execute
+    puts "", "Importing from Mongodb...."
+
+    import_users
+    import_discussions
+
+    puts "", "Done"
+  end
+
+  def import_users
+
+    puts '', "Importing users"
+
+    # fetch distinct list of Yahoo "profile" names
+    profiles = @collection.aggregate(
+                 [
+                  {"$group": { "_id": { profile: "$ygData.profile"  } } }
+                 ]
+            )
+
+    user_id = 0
+
+    create_users(profiles.to_a) do |u|
+
+      user_id = user_id + 1
+
+      # fetch last message for profile to pickup latest user info as this may have changed
+      user_info = @collection.find({"ygData.profile": u["_id"]["profile"]}).sort({"ygData.msgId":-1}).limit(1).to_a[0]
+
+      # Store user_id to profile lookup
+      @user_profile_map.store(user_info["ygData"]["profile"], user_id)
+
+      puts "User created: #{user_info["ygData"]["profile"]}"
+
+      user =
+       {
+        id: user_id,  # yahoo "userId" sequence appears to have changed mid forum life so generate this
+        username: user_info["ygData"]["profile"],
+        name: user_info["ygData"]["authorName"],
+        email: user_info["ygData"]["from"], # mandatory
+        created_at: Time.now
+      }
+      user
+    end
+
+    puts "#{user_id} users created"
+
+  end
+
+  def import_discussions
+    puts "", "Importing discussions"
+
+    topics_count = 0
+    posts_count = 0
+
+    topics = @collection.aggregate(
+                 [
+                  {"$group": { "_id": { topicId: "$ygData.topicId"  } } }
+                 ]
+    ).to_a
+
+    # for each distinct topicId found
+    topics.each_with_index do |t, tidx|
+
+      # create "topic" post first.
+      # fetch topic document
+      topic_post = @collection.find({"ygData.msgId": t["_id"]["topicId"]}).to_a[0]
+      next if topic_post.nil?
+
+      puts "Topic: #{tidx+1} / #{topics.count()}  (#{sprintf('%.2f', ((tidx+1).to_f / topics.count().to_f)*100)}%)  Subject: #{topic_post["ygData"]["subject"]}"
+
+      if topic_post["ygData"]["subject"].to_s.empty?
+        topic_title = "No Subject"
+      else
+        topic_title = topic_post["ygData"]["subject"]
+      end
+
+      topic = {
+        id: tidx+1,
+        user_id: @user_profile_map[topic_post["ygData"]["profile"]] || -1,
+        raw: topic_post["ygData"]["messageBody"],
+        created_at: Time.at(topic_post["ygData"]["postDate"].to_i),
+        cook_method: Post.cook_methods[:raw_html],
+        title: topic_title,
+        category: ENV['CATEGORY_ID'],
```

Comment:
The usage of this environment variable is quite hidden. I'd suggest documenting it at the beginning of the import script (see https://github.com/discourse/discourse/blob/master/script/import_scripts/mbox.rb#L6-L12).

Reply:
ok, documented at the start of the import script.

Reply:
commits squashed as well!